src/server.h

```diff
@@ -3081,6 +3103,8 @@ sds keyspaceEventsFlagsToString(int flags);
 #define MEMORY_CONFIG (1<<0) /* Indicates if this value can be loaded as a memory value */
 #define PERCENT_CONFIG (1<<1) /* Indicates if this value can be loaded as a percent (and stored as a negative int) */
 #define OCTAL_CONFIG (1<<2) /* This value uses octal representation */
+#define SLOT_MASK_SHIFT 48 /* Slot id is stored in high bits of the Scan API cursor, this number defines by how many bits it's shifted. */
+#define SLOT_MASK ((unsigned long long)(CLUSTER_SLOTS - 1) << SLOT_MASK_SHIFT)
```

Comment:
I think 48 bits should be enough, it's 2^16 times more than advertised max number of keys that Redis can hold. I like your suggestion of using LSB for slot instead of the cursor, it should indeed result in smaller nominal cursor values as bits will be shifted by 14 instead of 48 and cursor values are usually not that big.

Reply:
Although we "advertise" 2^32, we did make the needed changes to support 2^64 in several cases. We had some clusters in AWS with more than 2^32 keys. I think we should consider bumping that number to 2^48.